packages/app/src/explorer/utils.ts

```diff
@@ -18,10 +18,8 @@ export function needsNxBadge(
     return true;
   }
 
-  if (hasAttribute(entity, 'NX_class')) {
-    const attrValues = attrValuesStore.get(entity);
-    const nxClass = attrValues?.NX_class;
-
+  const nxClass = attrValuesStore.getSingle(entity, 'NX_class');
+  if (nxClass) {
```

Comment:
The `hasAttribute` check/optimisation is now done in `attrValuesStore.getSingle()`, so we don't have to worry about it here.